modules/ting_das/ting_das.module

```diff
@@ -0,0 +1,378 @@
+      '#type' => 'value',
+      '#value' => $entity->getId(),
+    );
+
+    $form['entity']['title'] = array(
+      '#type' => 'value',
+      '#value' => $entity->getTitle(),
+    );
+
+    $form['loaner_id'] = array(
+      '#type' => 'value',
+      '#value' => $creds['library_card_number'],
+    );
+
+    $form['name'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Name'),
+      '#default_value' => $profile ? $profile->{'field_' . $provider['module'] . '_name'}->value() : '',
+      '#required' => TRUE,
+    );
+
+    $form['mail'] = array(
+      '#type' => 'textfield',
+      '#title' => t('Mail address'),
+      '#default_value' => $profile ? $profile->{'field_' . $provider['module'] . '_mail'}->value() : '',
+      '#element_validate' => array(
+        'ting_das_mail_validate',
+      ),
+      '#required' => TRUE,
+    );
+
+    $function = $provider['module'] . '_user_branches';
+    $preferred_branch = $profile ? $profile->{'field_' . $provider['module'] . '_preferred_branch'}->value() : '';
+    $form += ding_reservation_default_options_branch('fbs_preferred_branch', $preferred_branch, $function($user));
+
+    $default_interest_period = ding_provider_invoke('reservation', 'default_interest_period');
+    $form += ding_reservation_interest_period_selector('interest_period', $default_interest_period, _ting_das_get_interest_periods());
+
+    if (isset($texts['gdpr'])) {
+      $form['gdpr'] = array(
+        '#prefix' => '<div class="form-item form-type-markup gdpr-text">',
+        '#markup' => check_markup($texts['gdpr']['value'], $texts['gdpr']['format']),
+        '#suffix' => '</div>',
+        '#weight' => isset($texts['placement']) ? $texts['placement'] : 10,
+      );
+    }
+
+    $form['submit'] = array(
+      '#type' => 'submit',
+      '#submit' => array('ting_das_order_form_submit'),
+      '#ajax' => array(
+        'callback' => 'ting_das_order_form_ajax_callback',
+      ),
+      '#value' => t('Order digital copy'),
+      '#attributes' => array(
+        'onclick' => 'wt.sendinfo({ linkId: "order_digital_copy" });',
+      ),
+      '#weight' => 15,
+    );
+  }
+  else {
+    $form['title'] = array(
+      '#markup' => '<div class="popupbar-title"><strong>' . t('Not resident in the municipality') . '</strong></div>',
+    );
+
+    $form['description'] = array(
+      '#prefix' => '<div class="form-item form-type-markup description">',
+      '#markup' => t('Only resident in the municipality is allowed to order digital copies.'),
+      '#suffix' => '</div>',
+    );
+  }
+
+  return $form;
+}
+
+/**
+ * Mail validation.
+ */
+function ting_das_mail_validate($element, $form_state) {
+  $email = $element['#value'];
+  if (!empty($email)) {
+    if (!ding_base_validate_email($email)) {
+      form_set_error($element['#name'], t('Please fill in a valid email'));
+    }
+  }
+}
+
+/**
+ * Submit handler for the order form.
+ */
+function ting_das_order_form_submit($form, &$form_state) {
+
+}
```

Comment:
This is empty. If that is intentional then please state that.

Reply:
Removed.